CHANGELOG.md

```diff
@@ -9,6 +9,11 @@ This project adheres to [Semantic Versioning](https://semver.org/).
 - Fix `SigSet` incorrect implementation of `Eq`, `PartialEq` and `Hash`
   ([#1946](https://github.com/nix-rust/nix/pull/1946))
 
+### Added
+
+- Added mutex interface.
```

Comment:
This is an important feature so I understand it may warrant more writing, but I'm just not sure what to write?

It's mostly self explanatory right? Its a mutex.